tools/toolchains/__init__.py

```diff
@@ -849,6 +849,21 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    def check_toolchain_path(self, name):
+        """Check if the path to toolchain is valid. Exit if not.
+
+        Positional arguments:
+        name -- the name of the toolchain, as used when creating compile
+                commands to execute. Valid key for TOOLCHAIN_PATHS.
+
+        No return value but causes a system exit of execution if the path
+        does not exist.
+        """
```

Comment:
I have seen the light! Going to try to add them to everything I modify in the future.